python/utils/OpaqueArguments.h

```diff
@@ -99,6 +99,14 @@ inline py::args simplifiedValidateInputArguments(py::args &args) {
       if (shape.size() != 1)
         throw std::runtime_error("Cannot pass ndarray with shape != (N,).");
 
+      // Vector size validation
+      size_t size = shape[0].cast<size_t>();
+      if (size == 0 || (size & (size - 1)) != 0) {
```

Comment:
This check is only valid if the argument is used as an initializer (state) for a register of qubits. That is, we are allowed to pass a vector of any length of floats to a kernel in general.